compiler/rustc_span/src/lib.rs

```diff
@@ -2249,14 +2249,14 @@ where
 /// been reported to the user, so no need to continue checking.
 #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
 #[derive(HashStable_Generic)]
-pub struct ErrorGuaranteed(());
+pub struct ErrorGuaranteed;
```

Comment:
that means that everyone can construct it, which is not good. not very guaranteed anymore

Reply:
Ok, I will change that commit to just add a comment explaining this.